scanspec/specs.py

```diff
@@ -303,6 +344,9 @@ class Concat(Spec):
             description="The right-hand Spec to Concat, midpoints will appear later"
         ),
     ]
+    gap: A[
+        bool, schema(description="If True, force a gap in the output at the join")
```

Comment:
I can see many specs needs to deal with the gapped and non-gapped cases explicitly like this. Is there a way we can make it part of the composition instead? I am imagining something like

```python
Concat(spec_a, Concat(GAP, spec_b))

# ...or

Concat(spec_a, gapped(spec_b))
```

Reply:
I think it might have to be